packages/compass-e2e-tests/smoketests/auto-update-from.ts

```diff
@@ -0,0 +1,57 @@
+import { execute } from '../installers/helpers';
+import type { Package } from '../installers/types';
+
+// TODO: move this since we'll use the same for testing TO
+function testInstalledApp(
+  pkg: Package,
+  appPath: string,
+  env: Record<string, string>
+): Promise<void> {
+  return execute(
+    'npm',
+    [
+      'run',
+      '--unsafe-perm',
+      'test-packaged',
+      '--workspace',
+      'compass-e2e-tests',
+      '--',
+      '--test-filter=auto-update',
+    ],
+    {
+      env: {
+        ...process.env,
+        HADRON_AUTO_UPDATE_ENDPOINT_OVERRIDE: 'http://localhost:8080',
+        AUTO_UPDATE_UPDATABLE: pkg.updatable.toString(),
+        COMPASS_APP_NAME: pkg.appName,
+        COMPASS_APP_PATH: appPath,
+        ...env,
+      },
+    }
+  );
+}
+
+export async function testAutoUpdateFrom(pkg: Package) {
+  // install the app
+  console.log(`installing ${pkg.packageFilepath}`);
+  const { appPath, uninstall } = await pkg.installer({
+    appName: pkg.appName,
+    filepath: pkg.packageFilepath,
+  });
+
+  console.log(appPath);
+
+  try {
+    // TODO: start the autoupdate server
```

Comment:
An actual TODO for this PR. We could also start the autoupdate server on the outside and just leave it running, but I think we'll want to run it with a different configuration once we get to the "autoupdate TO" tests.